cfn-resources/project/cmd/resource/resource.go

```diff
@@ -467,75 +485,75 @@ func readProjectSettings(client *mongodbatlas.Client, id string, currentModel *M
 }
 
 // Get difference in Teams
-func getChangeInTeams(currentTeams []ProjectTeam, oTeams []*mongodbatlas.Result) (newTeams []*mongodbatlas.ProjectTeam,
-	changedTeams []*mongodbatlas.ProjectTeam, removeTeams []*mongodbatlas.ProjectTeam) {
+func getChangeInTeams(currentTeams []ProjectTeam, oTeams []admin.TeamRole) (newTeams []admin.TeamRole,
+	changedTeams []admin.TeamRole, removeTeams []admin.TeamRole) {
 	for _, nTeam := range currentTeams {
 		if nTeam.TeamId != nil && len(*nTeam.TeamId) > 0 {
 			matched := false
 			for _, oTeam := range oTeams {
-				if nTeam.TeamId != nil && *nTeam.TeamId == oTeam.TeamID {
-					changedTeams = append(changedTeams, &mongodbatlas.ProjectTeam{TeamID: *nTeam.TeamId, RoleNames: nTeam.RoleNames})
+				if nTeam.TeamId != nil && oTeam.TeamId != nil && *nTeam.TeamId == *oTeam.TeamId {
+					changedTeams = append(changedTeams, admin.TeamRole{TeamId: nTeam.TeamId, RoleNames: nTeam.RoleNames})
 					matched = true
 					break
 				}
 			}
 			// Add to newTeams
 			if !matched {
-				newTeams = append(newTeams, &mongodbatlas.ProjectTeam{TeamID: *nTeam.TeamId, RoleNames: nTeam.RoleNames})
+				newTeams = append(newTeams, admin.TeamRole{TeamId: nTeam.TeamId, RoleNames: nTeam.RoleNames})
 			}
 		}
 	}
 
 	for _, oTeam := range oTeams {
-		if len(oTeam.TeamID) > 0 {
+		if oTeam.TeamId != nil && len(*oTeam.TeamId) > 0 {
 			matched := false
 			for _, nTeam := range currentTeams {
-				if nTeam.TeamId != nil && *nTeam.TeamId == oTeam.TeamID {
+				if nTeam.TeamId != nil && oTeam.TeamId != nil && *nTeam.TeamId == *oTeam.TeamId {
 					matched = true
 					break
 				}
 			}
 			if !matched {
-				removeTeams = append(removeTeams, &mongodbatlas.ProjectTeam{TeamID: oTeam.TeamID, RoleNames: oTeam.RoleNames})
+				removeTeams = append(removeTeams, admin.TeamRole{TeamId: oTeam.TeamId, RoleNames: oTeam.RoleNames})
 			}
 		}
 	}
 	return newTeams, changedTeams, removeTeams
 }
 
 // Get difference in ApiKeys
-func getChangeInAPIKeys(groupID string, currentKeys []ProjectApiKey, oKeys []mongodbatlas.APIKey) (newKeys, changedKeys, removeKeys []UpdateAPIKey) {
+func getChangeInAPIKeys(groupID string, currentKeys []ProjectApiKey, oKeys []admin.ApiKeyUserDetails) (newKeys, changedKeys, removeKeys []UpdateAPIKey) {
```

Comment:
```suggestion
func getChangeInAPIKeys(groupID string, currentKeys []ProjectApiKey, oKeys []*admin.ApiKeyUserDetails) (newKeys, changedKeys, removeKeys []UpdateAPIKey) {
```
nit: If you're passing a big struct as an argument **and the function does not change the value in the struct**, using a pointer is better because it saves space

Reply:
I do agree that using pointers for big struct parameters is good optimization as it avoids a complete copy of the struct. However, in this case where the argument is a slice, I do not see the benefit as a slice is already a reference-like type. This means that the elements contained within the slice are not being copied.

Reply:
I agree about optimization but also having the struct instead of the pointer as a param carries the message that you're not going to modify it.  You don't need to modify if you pass a pointer but you could.